### PR TITLE
Remove `.js` from `.umd` suffix

### PR DESCRIPTION
**Pull request by seanpdoyle**
https://github.com/emberjs/emberjs-build/pull/94#discussion_r28895145

### CODE REVIEW

**Comment by stefanpenner** [comment]
:+1:

**Comment by seanpdoyle** [comment]
@stefanpenner this is erroring for both this and `dag-map`

https://travis-ci.org/tildeio/rsvp.js/jobs/59900998#L165-L173

```
Build failed.
Merge error: file "lib/rsvp.umd" exists in /home/travis/build/tildeio/rsvp.js/tmp/funnel-dest_qYevCi.tmp and /home/travis/build/tildeio/rsvp.js/tmp/compile_modules-tmp_dest_dir-M7oSBn2y.tmp - pass option { overwrite: true } to mergeTrees in order to have the latter file win
Error: Merge error: file "lib/rsvp.umd" exists in /home/travis/build/tildeio/rsvp.js/tmp/funnel-dest_qYevCi.tmp and /home/travis/build/tildeio/rsvp.js/tmp/compile_modules-tmp_dest_dir-M7oSBn2y.tmp - pass option { overwrite: true } to mergeTrees in order to have the latter file win
    at /home/travis/build/tildeio/rsvp.js/node_modules/broccoli-merge-trees/index.js:49:21
    at lib$rsvp$$internal$$tryCatch (/home/travis/build/tildeio/rsvp.js/node_modules/broccoli-merge-trees/node_modules/promise-map-series/node_modules/rsvp/dist/rsvp.js:489:16)
```

Please advise

**Comment by seanpdoyle** [comment]
ping @stefanpenner

**Comment by stefanpenner** [comment]
the `umd` files should not end up in the final output. Something is allowing them to leak through, in this case the production section of the build is colliding. But the underlying problem is that `umd` file is purely input, not output.

**Comment by stefanpenner** [comment]
the correct solution is to fix `broccoli-es6-modules-transpiler` to not leak the `entry` file

a quick fix to this repo would be:

```
diff --git a/Brocfile.js b/Brocfile.js
index ddae8d4..80b6cba 100644
--- a/Brocfile.js
+++ b/Brocfile.js
@@ -37,7 +37,7 @@ var testBundle = browserify(merge([
   browserify: { entries: ['./test/index.js'] }
 });
 
-var dist = rsvp;
+var dist = find(rsvp, '!lib/rsvp.umd');
 
 env('production', function() {
   dist = merge([
```

File: Brocfile.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ var testBundle = browserify(merge([
   browserify: { entries: ['./test/index.js'] }
 });
 
-var dist = rsvp;
+var dist = find(rsvp, '!lib/rsvp.umd');
```
Comment (seanpdoyle) [inline Brocfile.js]: @stefanpenner no :game_die: https://travis-ci.org/tildeio/rsvp.js/jobs/60308798#L421

```
Building
Built project successfully. Stored in "/home/travis/build/tildeio/rsvp.js/tmp/class-tests_dist-S3frj9dZ.tmp".
not ok 1 Mocha - "./node_modules/.bin/mocha dist/test/browserify.js"
    ---
        stdout: >
            
        stderr: >
            
            /home/travis/build/tildeio/rsvp.js/dist/test/browserify.js:10708
                  promise = RSVP.reject(reason);
                                 ^
            TypeError: Object #<Object> has no method 'reject'
                at Suite.<anonymous> (/home/travis/build/tildeio/rsvp.js/dist/test/browserify.js:10708:22)
                at context.describe.context.context (/home/travis/build/tildeio/rsvp.js/node_modules/mocha/lib/interfaces/bdd.js:75:10)
```

**Pull request closed without merging**